conda_libmamba_solver/solver.py

```diff
@@ -1004,7 +1005,20 @@ def _notify_conda_outdated(
             # This check can be silenced with a specific option in the context or in quiet mode
             return
 
-        current_conda_prefix_rec = PrefixData(context.conda_prefix).get("conda", None)
+        # manually check base prefix since `PrefixData(...).get("conda", None) is expensive
+        # once prefix data is lazy this might be a different situation
+        current_conda_prefix_rec = None
+        conda_meta_prefix_directory = os.path.join(context.conda_prefix, "conda-meta")
+        with suppress(OSError):
+            if os.path.lexists(conda_meta_prefix_directory):
+                for filename in os.scandir(conda_meta_prefix_directory):
+                    if (
+                        filename.name.endswith(".json")
+                        and filename.name.rsplit("-", 2)[0] == "conda"
+                    ):
+                        with open(filename.path) as f:
+                            current_conda_prefix_rec = PrefixRecord(**json.load(f))
+                            break
```

Comment:
```suggestion
                for entry in os.scandir(conda_meta_prefix_directory):
                    if (
                        entry.is_file()
                        and entry.name.endswith(".json")
                        and entry.name.rsplit("-", 2)[0] == "conda"
                    ):
                        with open(entry.path) as entry_file:
                            current_conda_prefix_rec = PrefixRecord(**json.load(entry_file))
                            break
```

It's better to clarify that `os.scandir` returns a list of entries, not filenames, and you have to check whether the entry is a file or not.

I wonder what the `json.load` call might trigger in terms of exceptions, which would bubble up. It would be good to be a little more defensive about this.

Reply:
It might not matter for tiny-JSON but FYI json.loads(entry_file.read()) is faster than passing a file object to json.load()

Reply:
Ah good point, lemme update my suggestion

Reply:
done

Reply:
Thanks for this feedback everyone. Didn't know that scandir returned this much. In the future I think I will use this method much more heavily.